src/main/java/app/model/agents/Evasion/EvasionAgent.java

```diff
@@ -48,61 +59,94 @@ public Move move()
+        if(closestSeenGuard != null)
+        {
+            counter = 0;
+            if(closestGuard == null)
+            {
+                closestGuard = closestSeenGuard;
+                guardDirection = closestGuard.sub(position);
+            }
+            else if(closestSeenGuard.dist(position) < closestGuard.dist(position))
+            {
+                closestGuard = closestSeenGuard;
+                guardDirection = closestGuard.sub(position);
+            }
+        }
+        else
+            counter++;
     }
 
-    private Move moveRandomDirected()
+    private Move moveRandom()
     {
-        // randomness should be a number between 0 and 1
         int theta = (int) (Math.random() * 360);
         Vector randomDirection = new Vector(0, 1).rotate(theta);
 
-        Vector guardDirection = closestAgent(Type.GUARD);
+        if(!checkIfLegalMove | noWallDetected(randomDirection, maxSprint))
+            return new Move(randomDirection, randomDirection.scale(maxSprint));
+        else if(moveCounter < MAX_MOVES)
+        {
+            moveCounter++;
+            return moveRandom();
+        }
+        else
+            return new Move(direction, new Vector(0,0));
+    }
 
+    private Move moveRandomDirected(double randomness)
+    {
         if(guardDirection == null)
         {
-            // TODO this should flip to a new state
-            //guardInView = false;
-            return new Move(randomDirection, randomDirection.scale(maxSprint));
+            return moveRandom();
         }
+        // randomness should be a number between 0 and 1
+        int theta = (int) (Math.random() * 360);
+        Vector randomDirection = new Vector(0, 1).rotate(theta);
 
         Vector flippedDirection = guardDirection.rotate(180);
 
         Vector mixedDirection = flippedDirection.scale(1 - randomness).add(randomDirection.scale(randomness));
 
-        return new Move(guardDirection, mixedDirection.scale(maxSprint));
+        if(!checkIfLegalMove | noWallDetected(mixedDirection, maxSprint))
+            return new Move(guardDirection, mixedDirection.scale(maxSprint));
+        else if(moveCounter < MAX_MOVES)
+        {
+            moveCounter++;
+            randomness = Math.max(MAX_RANDOMNESS, randomness+0.05);
+            return moveRandomDirected(randomness);
+        }
+        else
+            return new Move(direction, new Vector(0,0));
     }
 
     public Move moveDirected()
     {
-
-        // need condition on this value being null
-        Vector guardDirection = closestAgent(Type.GUARD);
-
-        if(guardDirection == null)
-        {
-            // TODO this should flip to a new state
-            // guardInView = false;
-            return new Move(direction, new Vector());
-        }
-
         Vector flippedDirection = guardDirection.rotate(180);
 
         // Keep looking back to keep the agent in view
+        if(!checkIfLegalMove | noWallDetected(flippedDirection, maxSprint))
```

Comment:
Typo

Reply:
what is the type?

if checkIfLegalMove = true, than the if-statement checks the move
if checkIfLegalMove = false, than the move goes through regardsless, so doesn't check if the move is legal

or do you mean a spelling mistake I'm overlooking?